app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java

```diff
@@ -414,13 +448,26 @@ private void onDepictionsLoaded(List<IdAndCaptions> idAndCaptions){
      * The imageSpacer is Basically a transparent overlay for the SimpleDraweeView
      * which holds the image to be displayed( moreover this image is out of
      * the scroll view )
+     *
+     *
+     * If the image is sufficiently large i.e. the image height extends the view height, we reduce
+     * the height and change the width to maintain the aspect ratio, otherwise image takes up the
+     * total possible width and height is adjusted accordingly.
+     *
      * @param scrollWidth the current width of the scrollView
      */
     private void updateAspectRatio(int scrollWidth) {
         if (imageInfoCache != null) {
             int finalHeight = (scrollWidth*imageInfoCache.getHeight()) / imageInfoCache.getWidth();
             ViewGroup.LayoutParams params = image.getLayoutParams();
             ViewGroup.LayoutParams spacerParams = imageSpacer.getLayoutParams();
+            params.width=scrollWidth;
```

Comment:
Spaces according to https://google.github.io/styleguide/javaguide.html#s4.6-whitespace here and below